python/dolfinx/geometry.py

```diff
@@ -270,3 +270,32 @@ def compute_distance_gjk(
 
     """
     return _cpp.geometry.compute_distance_gjk(p, q)
+
+def determine_point_ownership(
+    mesh: Mesh,
+    points: npt.NDArray[np.floating], 
+    cells: typing.Optional[np.ndarray] = None,
+    padding: float = 0.0
+) -> PointOwnershipData:
+    """Build PointOwnershipData for a mesh-points pair.
+    First, potential collisions are found
+    by building BoundingBoxTrees for the points
+    and the mesh cells and finding which leaves collide.
+    Then, actual cell-point containment pairs are determined
+    using the GJK algorithm.
+
+    Args:
+        mesh: The mesh
+        points: The points
+        cells: Cells in mesh who potentially contain the points.
+            If ``None`` then all cells are considered.
+        padding: Padding added to the bounding boxes.
+
+    Returns:
+        PointOwnershipData
+
+    """
+    if cells is None:
+        map = mesh.topology.index_map(mesh.topology.dim)
+        cells = np.arange(map.size_local + map.num_ghosts, dtype=np.int32)
```

Comment:
It should be only `map.size_local` here but I don't know how to modify a pull request.